### PR TITLE
E2e test

### PR DESCRIPTION
**Pull request by mayerm94**
Add end to end integration testing. Fix docker-compose.override.yml so that it will copy certificates to the endpoint_querier so that it will start behind a ssl-inspecting proxy.

Pull requests into this repository require the following checks to be completed by the submitter and reviewers.

**Submitter:**
- [ ] Administrative tasks are complete.
  - This pull request describes why these changes were made.
  - The JIRA ticket links to this PR.
  - The JIRA ticket for this PR is: https://oncprojectracking.healthit.gov/support/browse/LANTERN-94
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [ ] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.
- [ ] Code review has been performed.
  - Code diff has been done and checked to ensure only expected code is being committed.
  - A linter for the language has been run.

**Primary Reviewer:** @edeyoung 

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [x] Code review has been performed.
  - Code is maintainable, reusable, efficient, and correct.
  - Code accomplishes the tasks purpose.
  - Code follows style guidance appropriate for the language, including passing linter checks.
  - Code is well commented (which does not mean verbosely commented).
  - vendor/ directories have been updated
    - New dependencies have been added
    - Local dependencies have been updated
- [x] Tests are complete.
  - Tests are included and test edge cases.
  - Tests have been run locally and pass.

**Secondary Reviewer:**

- [ ] Code works.
  - Code has been run locally and the basic functionality affected by the PR has been checked.
  - Edge cases have been checked to ensure behavior is as expected.
- [ ] Code review has been performed.
  - Code accomplishes the tasks purpose.
- [ ] Tests are complete.
  - Tests have been run locally and pass.

**Merger:**

Prior to merging, ensure that all checks have been completed. Delete the branch after the merge is complete.

### CODE REVIEW

File: e2e/wait-for-it.sh [hunk 1]
```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env bash
```
Comment (edeyoung) [inline e2e/wait-for-it.sh]: do we need to cite this script?

File: .travis.yml [hunk 2]
```diff
@@ -24,4 +24,5 @@ before_script:
 
 script:
   - golangci-lint run -E gofmt
-  - go test -v -race ./...  # Run all the tests with the race detector enabled
\ No newline at end of file
+  - go test -v -race ./...  # Run all the tests with the race detector enabled
+  - docker-compose -f docker-compose.yml -f docker-compose.test.yml up --build --abort-on-container-exit
```
[the remark this answers is not in the document]
Comment (mayerm94) [reply .travis.yml]: b

File: README.md [hunk 3]
```diff
@@ -140,6 +142,12 @@ If you stopped the containers and wish to restart them you can run:
 docker-compose start
 ```
 
+## Starting Services Behind SSL-Inspecting Proxy
+If you are operating behind a proxy that does SSL-Inspection you will have to copy the certificates that are used by the proxy into the docker containers that will be communicating through the proxy. Currently the endpoint_querier is the only contianer that has such a requirement, the volumes entry `- ./certs/:/etc/ssl/certs` in the endpoint_querier service section  of `docker-compose.override.yml` will mount a directory named `certs` located in the root of this project into the location of the docker container where the container's OS will look for certificates. If you are operating behind an SSL-Inspecting proxy **you will have to copy your certificates into this directory.** In order for the changes in the `docker-compose.override.yml` file to be applied you will have to specify the override file when you run docker-compose.
```
[the remark this answers is not in the document]
Comment (mayerm94) [reply README.md]: true. will change

File: e2e/wait-for-it.sh (hunk 1 above)
Comment (mayerm94) [reply e2e/wait-for-it.sh]: Probably a good idea --done

**Pull request closed without merging**